web-admin/src/features/errors/error-utils.ts

```diff
@@ -1,57 +1,74 @@
+export function createErrorCallback(queryClient: QueryClient) {
+  return (error: AxiosError) => {
+    const isProjectPage = get(page).route.id === "/[organization]/[project]";
+    const isDashboardPage =
+      get(page).route.id === "/[organization]/[project]/[dashboard]";
 
-  if (!error.response) return;
+    if (!error.response) return;
 
-  // Special handling for some errors on the Project page
-  if (isProjectPage) {
-    // If "repository not found", ignore the error and show the page
-    if (
-      error.response?.status === 400 &&
-      (error.response.data as RpcStatus).message === "repository not found"
-    ) {
-      return;
+    // Special handling for some errors on the Project page
+    if (isProjectPage) {
+      // If "repository not found", ignore the error and show the page
+      if (
+        error.response?.status === 400 &&
+        (error.response.data as RpcStatus).message === "repository not found"
+      ) {
+        return;
+      }
     }
-  }
 
-  // Special handling for some errors on the Dashboard page
-  if (isDashboardPage) {
-    // If a dashboard wasn't found, let +page.svelte handle the error.
-    // Because the project may be reconciling, in which case we want to show a loading spinner not a 404.
-    if (
-      error.response?.status === 404 &&
-      (error.response.data as RpcStatus).message === "not found"
-    ) {
-      return;
+    // Special handling for some errors on the Dashboard page
+    if (isDashboardPage) {
+      // If a dashboard wasn't found, let +page.svelte handle the error.
+      // Because the project may be reconciling, in which case we want to show a loading spinner not a 404.
+      if (
+        error.response?.status === 404 &&
+        (error.response.data as RpcStatus).message === "not found"
+      ) {
+        return;
+      }
+
+      // When a JWT doesn't permit access to a metrics view, the metrics view APIs return 401s.
+      // In this scenario, `GetCatalog` returns a 404. We ignore the 401s so we can show the 404.
+      if (error.response?.status === 401) {
+        return;
+      }
     }
 
-    // When a JWT doesn't permit access to a metrics view, the metrics view APIs return 401s.
-    // In this scenario, `GetCatalog` returns a 404. We ignore the 401s so we can show the 404.
+    // If Unauthorized, redirect to login page
     if (error.response?.status === 401) {
+      goto(`${ADMIN_URL}/auth/login?redirect=${window.origin}`);
       return;
     }
-  }
 
-  // If Unauthorized, redirect to login page
-  if (error.response?.status === 401) {
-    goto(`${ADMIN_URL}/auth/login?redirect=${window.origin}`);
-    return;
-  }
+    if (
+      error.response?.status === 400 &&
+      (error.response.data as RpcStatus).message === "driver: not found" &&
+      (isProjectPage || isDashboardPage)
+    ) {
+      const [, org, proj] = get(page).url.pathname.split("/");
+      // goto(`/${org}/${proj}/-/status`);
```

Comment:
Old code?

Reply:
I guess i left it out as a question on where we should be redirecting to. The status page or the dashboard list page.